### PR TITLE
<FEAT>Implement the duration estimation ratio value

### LINKED ISSUE #224: [FEAT] Implement the duration estimation ratio value
### Requested Feature/Enhancement

The default parameter duration_estimate.duration_factor is currently a placeholder. 

### Need for feature/Enhancement

Implement the duration_estimate.duration_factor functionality.
This parameter was intended to allow users to provide a ratio to use as the duration estimation. Currently, it's always set to use the full duration between surveys. 

### Problem to solve

_No response_

### Implementation Ideas

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by SEAJang**
# Pull Request Key Information

## Reason for change

The duration estimation ratio existed as a placeholder and functionally did not do anything

## What was changed

The functionality for duration estimation ratio was implemented

## Intended Purpose

Allow users to customize the duration estimations used by the uncertainty measurements

## Level of version change required

Minor

## Testing Completed

New unit tests were added
All  unit tests pass
[results.txt](https://github.com/user-attachments/files/16156415/results.txt)

Manual testing was completed - specifically looking at when the ratio is set to 0.0 and 0.5 - 0.0 exaggerated the existing bugs significantly. 

## Target Issue

Closes #224 

## Additional Info

Several warnings have been introduced with the update of the pandas library. A new issue will be created to address these warnings. 
Tagging to update this to V4.1.0 will happen after this PR has been approved.

### CODE REVIEW

**Comment by SEAJang** [comment]
[results.txt](https://github.com/user-attachments/files/16156509/results.txt)
Unit test results after rebasing on the current head of master.

File: USER_MANUAL.md [hunk 1]
```diff
@@ -866,17 +866,17 @@ It is important to note that LDAR-Sim internally converts KG of methane to MMBtu
...
 
 **Data Type:** Numeric
 
-**Default input:** 1
+**Default input:** 1.0
 
 **Description:** A decimal number representing the ratio of time since the last survey or screening at a given site, used to estimate the duration of a given measurement.
 
-For example, a value of 0.5 means that all measurements are based on the assumption that half the time since the last survey or screening at a given site is used to estimate the duration of the measured emission.
+For example, a value of 0.6 means that 60% of the duration was assumed to be the greater emitting value, between the two measurement points, and 40% of the duration would be assuming the smaller of the measured emission.
```
Comment (ThomasGalesloot) [inline USER_MANUAL.md:877]: Thoughts on including an timeseries image here as a visual resource?

File: changelog.md [hunk 2]
```diff
@@ -1,5 +1,9 @@
 # Change Log
 
+## 2024-07-11 - Version 4.1.0
```
Comment (ThomasGalesloot) [inline changelog.md:3]: I think one other change made it in after version 4.0.4

**Review by ThomasGalesloot — APPROVED** [review]
See my comments before merging but great work!

File: LDAR_Sim/src/file_processing/output_processing/program_output_helpers.py [hunk 3]
```diff
@@ -96,3 +161,51 @@ def expand_column(df: pd.DataFrame, column: str) -> pd.DataFrame:
     combined_df = combined_df.loc[:, ~combined_df.columns.duplicated()]
 
     return combined_df
+
+
+def calculate_new_start_date(start_date, end_date, duration_factor):
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/src/file_processing/output_processing/program_output_helpers.py:166]: Forgot to remove this code when the code using it was removed.

File: LDAR_Sim/testing/unit_testing/test_file_processing/test_output_processing/test_program_output_helpers/test_determine_prev_date.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+            ),
+            "value": [1, 2, 3, 4, 5],
+        }
+    )
+    result: pd.Series = determine_prev_date(test_input)
+    # Expected output
+    expected_output = pd.Series(
+        pd.to_datetime(
+            [
+                "2021-01-01",
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/testing/unit_testing/test_file_processing/test_output_processing/test_program_output_helpers/test_determine_prev_date.py:48]: The fillna will replace the NaN that would result from the shift with the original value.

File: LDAR_Sim/testing/unit_testing/test_file_processing/test_output_processing/test_program_output_helpers/test_determine_next_date.py [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    result: pd.Series = determine_next_date(test_input)
+    # Expected output
+    expected_output = pd.Series(
+        pd.to_datetime(
+            [
+                "2021-01-02",
+                "2021-01-03",
+                "2021-01-04",
+                "2021-01-05",
+                "2021-01-05",
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/testing/unit_testing/test_file_processing/test_output_processing/test_program_output_helpers/test_determine_next_date.py:52]: The fillna will replace the NaN that would result from the shift with the original value.

File: changelog.md (hunk 2 above)
Comment (SEAJang) [reply changelog.md:3]: Will address with a rebase.

File: LDAR_Sim/testing/unit_testing/test_file_processing/test_output_processing/test_program_output_helpers/test_calculate_new_start_date.py [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+
+from file_processing.output_processing.program_output_helpers import calculate_new_start_date
+
+
+def half_duration():
+    return ([date(2021, 1, 1), date(2021, 1, 30), 0.5], date(2021, 1, 15))
+
+
+def failing_case1():
+    return ([date(2021, 1, 1), date(2021, 1, 30), 0], ValueError)
```
[the remark this answers is not in the document]
Comment (SEAJang) [reply LDAR_Sim/testing/unit_testing/test_file_processing/test_output_processing/test_program_output_helpers/test_calculate_new_start_date.py:32]: No longer relevant.

**Review by ThomasGalesloot — APPROVED** [review]

File: USER_MANUAL.md (hunk 1 above)
Comment (SEAJang) [reply USER_MANUAL.md:877]: I don't think this is necessary.

**Pull request merged**